SECEdgar/base.py

```diff
@@ -83,6 +92,8 @@ def _handle_error(response):
         Raises:
             EDGARQueryError: If response returns error code.
         """
+        if response is None:
```

Comment:
Our network layer is also becoming complicated. We can refactor this by having a `NetworkClient` class. The client will have the responsibility to create the request, trigger the request, parse the data and also error handling.

And this client will be used by `_EDGARBASE`. 

Currenlty `_EDGARBase` is suppose to work as a abstract class but it has been laiden with lot of implementation details.

Reply:
Again, we can move this to separate ticket.

Reply:
I think this is a good idea. I will see if I can find other projects who have implemented a similar solution to get a better sense of how this should be structured.

Reply:
Thank you :)